blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubRepositories.java

```diff
@@ -64,7 +65,8 @@ public GithubRepositories(StandardUsernamePasswordCredentials credentials, Strin
                     parent.getRepoType(),
                     pageSize, pageNumber), accessToken);
 
-            this.repositories = GithubScm.om.readValue(HttpRequest.getInputStream(connection), GH_REPO_EX_LIST_TYPE);
+            this.repositories = GithubScm.getMappingObjectReader().forType(GH_REPO_EX_LIST_TYPE)
+                .readValue(HttpRequest.getInputStream(connection));
```

Comment:
Just curious, what would the right approach look like? IIUC, Blue Ocean implemented its own HTTP layer, but is trying to use the data binding layer from github-api, and instead it should be using a higher-level API that does both at once?

Reply:
Correct, BO shouldn't have reimplemented the only HTTP layer. It should be using the github-api or entirely implementing its own data mapping.   

The objects BO binds are only valid as data containers and even some getters will not work correctly due to the objects having no reference to the expected `GitHub` connection.  

Also, BO doesn't use OkHttp so it gets none of the benefits of any caching. 

I'm sure there were good reasons for this being implemented this way, but it is really unfortunate. 

Whatever behavior was missing could certainly have been added to github-api and in fact most of it has (and even been improved on).  But going through and converting all the places HTTPRequest is used will take more time and have more risk of further breakage.  

This method is a prefect example - it reimplements the paged reading that is already in github-api (including [all the page url parsing](https://github.com/jenkinsci/blueocean-plugin/pull/2064/files#diff-cd796a86ce13b64cde1471f6a4165551R77-R97) ).

Reply:
That would be great but so much big change... :(